src/integration.rs

```diff
@@ -325,176 +335,332 @@ mod tests {
+
+            prepare_tap(&net);
+
+            let mut c = Command::new("qemu-system-x86_64");
+            c.args(&[
+                "-machine",
+                "q35,accel=kvm",
+                "-cpu",
+                "host,-vmx",
+                fw.fw_type,
+                fw.path,
+                "-display",
+                "none",
+                "-nodefaults",
+                "-serial",
+                "stdio",
+                "-drive",
+                &format!("id=os,file={},if=none", os),
+                "-device",
+                "virtio-blk-pci,drive=os,disable-legacy=on",
+                "-m",
+                "4G",
+                "-netdev",
+                &format!(
+                    "tap,id=net0,ifname={},script=no,downscript=no",
+                    net.tap_name
+                ),
+                "-device",
+                &format!("virtio-net-pci,netdev=net0,mac={}", net.guest_mac),
+            ]);
+
+            let stdout = fs::File::create(tmp_dir.path().join("stdout")).unwrap();
+            let stderr = fs::File::create(tmp_dir.path().join("stderr")).unwrap();
+
+            eprintln!("Spawning: {:?}", c);
+            let mut child = c
+                .stdout(Stdio::from(stdout))
+                .stderr(Stdio::from(stderr))
+                .spawn()
+                .expect("Expect launching QEMU to succeed");
+
+            thread::sleep(std::time::Duration::from_secs(60));
+            let r = std::panic::catch_unwind(|| {
+                let auth = windows_auth();
+                ssh_command_with_auth(&net.guest_ip, "shutdown /s", &auth)
+                    .expect("Expect SSH command to work");
+            });
+
+            child.kill().unwrap();
+            let output = child.wait_with_output().unwrap();
+
+            cleanup_tap(&net);
+
+            handle_child_output(&tmp_dir, r, &output);
+        }
+
+        #[test]
+        #[cfg(not(feature = "coreboot"))]
+        fn test_boot_qemu_windows() {
+            let fw = Firmware {
+                fw_type: "-kernel",
+                path: "target/target/release/hypervisor-fw",
+            };
+            test_boot_qemu_windows_common(&fw);
+        }
+
+        #[test]
+        #[cfg(feature = "coreboot")]
+        fn test_boot_qemu_windows() {
+            let fw = Firmware {
+                fw_type: "-bios",
+                path: "resources/coreboot/coreboot/build/coreboot.rom",
+            };
+            test_boot_qemu_windows_common(&fw);
+        }
+
+        #[test]
+        #[cfg(not(feature = "coreboot"))]
+        fn test_boot_ch_windows() {
+            let tmp_dir = TempDir::new().expect("Expect creating temporary directory to succeed");
+            let os = prepare_os_disk(&tmp_dir, WINDOWS_IMAGE_NAME);
+
+            let mut c = Command::new("./resources/cloud-hypervisor");
+            c.args(&[
+                "--cpus",
+                "boot=2,kvm_hyperv=on",
```

Comment:
Do you need the equivalent of `kvm_hyperv=on` for QEMU?

Reply:
As far as I tested, no Windows-specific option is required for coreboot/QEMU, but I didn't test the Windows guest with QEMU `-kernel` option mode. I think this test combination cannot be supported for now.